### ISSUE TITLE
Spotify with ALSA cannot play songs under firejail

### ISSUE THREAD

**Issue by Tritonio** [netblue30/firejail#1872]
I had issues with PulseAudio and today I uninstalled it. Everything is running fine with ALSA now: Chromium (in firejail), VLC, system sounds etc. But Spotify, when run under firejail, will not play any songs. If I start it from a console like "firejail spotify I get the following:

```
Reading profile /etc/firejail/spotify.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 6023, child pid 6024
Private /etc installed in 7.52 ms
Warning: skipping spotify for private /opt
Private /opt installed in 0.03 ms
Blacklist violations are logged to syslog
Child process initialized in 55.43 ms
[0408/232919.825842:ERROR:gl_implementation.cc(245)] Failed to load libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[0408/232919.914421:ERROR:gpu_child_thread.cc(252)] Exiting GPU process due to errors during initialization
[0408/232919.923520:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
[0408/232919.996605:ERROR:gl_implementation.cc(245)] Failed to load /usr/share/spotify/swiftshader/libGLESv2.so: /usr/share/spotify/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[0408/232919.998298:ERROR:gpu_child_thread.cc(252)] Exiting GPU process due to errors during initialization
[0408/232920.069995:ERROR:gl_implementation.cc(245)] Failed to load /usr/share/spotify/swiftshader/libGLESv2.so: /usr/share/spotify/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[0408/232920.071566:ERROR:gpu_child_thread.cc(252)] Exiting GPU process due to errors during initialization
[0408/232920.142997:ERROR:gl_implementation.cc(245)] Failed to load /usr/share/spotify/swiftshader/libGLESv2.so: /usr/share/spotify/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[0408/232920.144526:ERROR:gpu_child_thread.cc(252)] Exiting GPU process due to errors during initialization
[0408/232920.216233:ERROR:gl_implementation.cc(245)] Failed to load /usr/share/spotify/swiftshader/libGLESv2.so: /usr/share/spotify/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[0408/232920.217780:ERROR:gpu_child_thread.cc(252)] Exiting GPU process due to errors during initialization
[0408/232920.222172:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
[0408/232920.223605:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
[0408/232920.225841:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
[0408/232920.226867:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
[0408/232920.227323:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
ALSA lib pcm_direct.c:1679:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)

```

There are no messages in /var/log/syslog related to ALSA or Spotify (only one message about spotify not using the notifications correctly)

When I had PulseAudio, Spotify was working fine under firejail.

I am using firejail version 0.9.52 on Mint with the MATE desktop.

**Comment by Fred-Barclay** [comment]
G'day @Tritonio 
Can you edit /etc/firejail/spotify.profile and replace the line beginning with `private-etc` with the following line:
```
private-etc group,fonts,ld.so.cache,machine-id,pulse,resolv.conf
```

Thanks!
Fred

**Comment by Tritonio** [comment]
Hello Fred! You're the best. And actually it doesn't need all of these. Just adding "group" was enough:

> private-etc group,fonts,machine-id,pulse,resolv.conf

Should I assume that "The field ipc_gid must be a valid group (create group audio)" is what gave it away for you? I'll check the documentation for "private-etc" to see what it does.

Thanks again!

**Comment by Fred-Barclay** [comment]
G'day @Tritonio I actually had a similar experience a few weeks ago when installing Arch (see https://github.com/netblue30/firejail/commit/a2aaf6fb3829a2407ea4cb1329819c8c5e5d12ca). Spotify wouldn't work without `groups` in `private-etc`, so I added a commit, only to never be able to replicate the issue (that is, Spotify started working without `groups`). I ended up reverting that change. 

Your post here tipped me off that alsa must require `groups` in the private-etc filter, at least for Spotify.

Cheers! I'll (re)fix this right now. :smile: 
Fred
